pycaption/dfxp/base.py

```diff
@@ -124,17 +126,35 @@ def _find_times(self, p_tag):
         return start, end
 
     def _translate_time(self, stamp):
-        timesplit = stamp.split(u':')
-        if u'.' not in timesplit[2]:
-            timesplit[2] = timesplit[2] + u'.000'
-        secsplit = timesplit[2].split(u'.')
-        if len(timesplit) > 3:
-            secsplit.append((int(timesplit[3]) / 30) * 100)
-        microseconds = (int(timesplit[0]) * 3600000000 +
-                        int(timesplit[1]) * 60000000 +
-                        int(secsplit[0]) * 1000000 +
-                        int(secsplit[1]) * 1000)
-        return microseconds
+        if stamp[-1].isdigit():
+            timesplit = stamp.split(u':')
+            if u'.' not in timesplit[2]:
+                timesplit[2] = timesplit[2] + u'.000'
+            secsplit = timesplit[2].split(u'.')
+            if len(timesplit) > 3:
+                secsplit.append((int(timesplit[3]) / 30) * 100)
+            microseconds = (int(timesplit[0]) * 3600000000 +
+                            int(timesplit[1]) * 60000000 +
+                            int(secsplit[0]) * 1000000 +
+                            int(secsplit[1]) * 1000)
+            return microseconds
+        else:
+            # Must be offset-time
+            m = re.search('^([0-9.]+)([a-z]+)$', stamp)
+            value = float(m.group(1))
+            metric = m.group(2)
+            if metric == u"h":
+                microseconds = value * 60 * 60 * 1000000
+            elif metric == u"m":
+                microseconds = value * 60 * 1000000
+            elif metric == u"s":
+                microseconds = value * 1000000
+            elif metric == u"ms":
+                microseconds = value * 1000
+            else:
+                raise InvalidInputError(u"Unsupported offset-time metric " + metric)
+
+            return microseconds
```

Comment:
Although we lack support for offset time atm, I see no tests for this. Please add some. You can either unit-test this method, or make a more integration-like test, using input and output samples.